src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,290 @@ void CmdSketcherViewSection::activated(int iMsg)
 
 bool CmdSketcherViewSection::isActive()
 {
-    Gui::Document *doc = getActiveGuiDocument();
-    if (doc) {
-        // checks if a Sketch Viewprovider is in Edit and is in no special mode
-        SketcherGui::ViewProviderSketch* vp = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
-        if (vp /*&& vp->getSketchMode() == ViewProviderSketch::STATUS_NONE*/)
-            return true;
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid tool ------------------------------ Grid Activate */
+DEF_STD_CMD_A(CmdSketcherGridActivate)
+
+bool toggleGrid(Gui::Document* doc) {
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    bool newState = ! hGrp->GetBool("ShowGrid");
+
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+    if (sketchView) {
+        hGrp->SetBool("ShowGrid", newState);
+        sketchView->ShowGrid.setValue(newState);
     }
-    return false;
+    return newState;
+}
+
+CmdSketcherGridActivate::CmdSketcherGridActivate()
+    : Command("Sketcher_GridActivate")
+{
+    sAppModule = "Sketcher";
+    sGroup = "Sketcher";
+    sMenuText = QT_TR_NOOP("Activate Grid");
+    sToolTipText = QT_TR_NOOP("Activate and show the sketcher grid.");

```

Comment:
The command name and its description is misleading because in fact it toggles the grid, i.e. switches it on or off.

Reply:
I'm not sure to understand. Activate means switching on/off ? What name would you prefer?

Reply:
Activate means on, probably should be "Toggle Grid"

Reply:
Oh ok I see.